File: devops-scm-api/src/main/java/com/tencent/devops/scm/api/pojo/Issue.java
package com.tencent.devops.scm.api.pojo;

import java.util.Date;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Issue {
    private Long id;
    private Integer number;
    private String title;
    private String body;
    private String link;
    private List<String> labels;
    private Boolean closed;
    private Boolean locked;
    private User author;
    private Date created;
    private Date updated;
    private String milestoneId;
    private String state;
}

File: devops-scm-api/src/main/kotlin/com/tencent/devops/scm/api/pojo/Issue.kt
package com.tencent.devops.scm.api.pojo

import java.time.LocalDateTime

data class Issue(
    var id: Long,
    var number: Int,
    var title: String,
    var body: String? = null,
    var link: String,
    var labels: List<String>? = null,
    var closed: Boolean = false,
    var locked: Boolean = false,
    var author: User,
    var created: LocalDateTime,
    var updated: LocalDateTime? = null,
    var milestoneId: String? = null,
    var state: String? = null
)
